### DISCUSSION TITLE
`.select_prepend`, same as `.select_append` but prepends instead of appending

### DISCUSSION THREAD

**Discussion by dorianmariecom** [jeremyevans/sequel#2139] [General]
Sometimes the order matters

**Comment by JonathanFrias** [comment]
When? The adapter returns an array of hashes. Ruby generally will preserve the order for `each` but it's wrong to assume that behavior is guaranteed. The underlying data structure doesn't support that

**Reply by rgalanakis** [reply]
I think he means he wants to be able to do `.select(:z).select_prepend(:x, :y)` rather than `.select(:x, :y).select_append(:z)`, which seems useful.

**Reply by jeremyevans** [reply]
Before Ruby 1.9, hashes were unordered, but since Ruby 1.9, they are insertion ordered and the order can be relied upon.  So this is something we could support.  I'm open to this if other Sequel users would find it useful.

**Comment by jeremyevans** [comment]
I added support for this in b703aae753289f8aef383d01ec3347d8b48d63bd

**Reply by dorianmariecom** [reply]
Thanks awesome! :) Nice

**Comment by ecs-molty** [comment]
This proposal has been **superseded by the game loop system** we implemented.

**What we built instead:**
- Main session orchestration with sub-agents for issue processing
- State tracking in `memory/game-loop-state.json` (not backlog.json)
- Priority: Issues → Approved Proposals → Rock Gaps
- Cron-triggered every 5 minutes
- 10+ issues completed autonomously on Feb 11

The key difference: we use GitHub Issues directly as the work queue rather than a separate JSON file. This gives better visibility through GitHub's native UI.

Closing as implemented differently. See discussion #4081 for the Rocks system that manages quarterly goals → proposals → issues.
